src/main/java/openblocks/common/entity/EntityItemProjectile.java

```diff
@@ -37,66 +37,21 @@ public static void registerFixes(DataFixer fixer) {
 		fixer.registerWalker(FixTypes.ENTITY, new ItemStackData(EntityItemProjectile.class, "Item"));
 	}
 
+	private boolean firstUpdate = true;
+
 	@Override
 	public void onUpdate() {
-		final double x = posX;
-		final double y = posY;
-		final double z = posZ;
-
-		final double vx = motionX;
-		final double vy = motionY;
-		final double vz = motionZ;
+		// Remove the air drag that EntityItem.onUpdate adds to our velocity
+		if (!firstUpdate && !this.onGround) {
+			float f = 0.98F;
+			this.motionX = this.motionX / f;
+			this.motionY = this.motionY / f;
+			this.motionZ = this.motionZ / f;
+		}
+		firstUpdate = false;
 
 		// let vanilla run
 		super.onUpdate();
-		if (!isDead) return;
```

Comment:
It was supposed to be `isDead`, then whole thing becomes more sensible.

Basically, purpose of this code was to let `super.onUpdate` handle all state changes, then overwrite movement with our own. Restoring just motion before update leads to numerical errors, which would cause mismatch with canon targeting algorithm (not that it was super precise in first place though).

Reply:
Yea, although with `isDead` instead it was double-moving the item, so that would have to get fixed too. It's basically a choice between the way I've got it, which is a a bit of a hack no doubt, and duplicating half the code inside EntityItem.onUpdate. If you'd prefer it the other way, that's fine by me. I wouldn't worry about the numerical errors though--these values are all doubles pretty close to 1, so back-of-the-envelope calculation says worst case is an acceleration of about 4.5e-15 blocks/s^2.